src/Network/OAuth/Simple.hs

```diff
@@ -122,22 +118,18 @@ upgradeCred tok = liftM (Cred.upgradeCred tok . cred) ask
 -- with the same configuration but new credentials.
 upgrade :: (Cred.ResourceToken ty', Monad m) => O.Token ty' -> OAuthT ty' m a -> OAuthT ty m a
 upgrade tok oat = do
-  gen  <- state R.cprgFork
```

Comment:
PR #17 also deletes this line, but then tried to simulate it, or it did prior to discussion (see https://github.com/tel/oauthenticated/pull/17#commitcomment-11676688). I'm not exactly sure what if any of that is needed/possible here since the state of the generator is no longer under the control of this code.